### PR TITLE
prompts for the conv_ai dataset

### PR DESCRIPTION
**Pull request by josrozen**
6 prompts around engagement in conversations given a context

### CODE REVIEW

File: templates/conv_ai/templates.yaml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      |||
+
+      {% set alice = (evaluation|selectattr("userId", "equalto", "Alice")|first)["engagement"]
+      %}
+
+      {% set bob = (evaluation|selectattr("userId", "equalto", "Bob")|first)["engagement"]
+      %}
+
+
+      {% if alice == bob %}both{% elif alice < bob %}Bob{% else %}Alice{% endif %}'
```
Comment (VictorSanh) [inline templates/conv_ai/templates.yaml:82]: this template should be conditional: it gives a very weird prompted example where the thread has only 1 or 2 utterances (or even 0)...
Capture d’écran 2021-06-15 à 5 12 25 PM
```suggestion
    jinja: '{% if thread | length >= 4%}
    {{context}}


      Given the previous context, who do you think is more engaged in this conversation
      (Alice, Bob, or both):

      {% for utterance in thread %}

      - {{ utterance["userId"] }}: {{ utterance["text"] }}

      {% endfor %}

      |||

      {% set alice = (evaluation|selectattr("userId", "equalto", "Alice")|first)["engagement"]
      %}

      {% set bob = (evaluation|selectattr("userId", "equalto", "Bob")|first)["engagement"]
      %}


      {% if alice == bob %}both{% elif alice < bob %}Bob{% else %}Alice{% endif %}
      {% endif %}'
```

**Review by VictorSanh — APPROVED** [review]
this is such a strange dataset
please verify my suggestion, otherwise lgtm!

File: templates/conv_ai/templates.yaml (hunk 1 above)
Comment (josrozen) [reply templates/conv_ai/templates.yaml:82]: it does look weird. I should update all templates though. I'd suggest using something like ```{% if (0 < (thread | selectattr("userId", "equalto", "Bob") | list | length)) and (0 < (thread | selectattr("userId", "equalto", "Alice") | list | length)) %}```. Would that be ok?

Comment (VictorSanh) [reply templates/conv_ai/templates.yaml:82]: yeah your proposition looks good and should cover much of the cases I was worried about!
i think that would make sense to update all templates cause not sure whether it is useful who's the most engaged when someone is talking alone....

Comment (josrozen) [reply templates/conv_ai/templates.yaml:82]: there's still a (minor?) issue if I do this: the tests don't like the setting of alice and bob variables (```{% set alice = (evaluation|selectattr("userId", "equalto", "Alice")|first)["engagement"]```) inside of the all surrounding ```if``` and complain about "unrecognized" variables. This means I have to move the 2 ```set```at the top of the prompt, outside the ```if```. It works but will it correctly skip the incomplete examples we're trying to filter out?

**Comment by josrozen** [comment]
thanks @VictorSanh, I've added the surrounding tests to exclude edge examples lacking some utterances from a participant.

File: templates/conv_ai/templates.yaml (hunk 1 above)
Comment (VictorSanh) [reply templates/conv_ai/templates.yaml]: yes you'll have to do `set alice` and `set bob` outside of the if condition indeed!

The example will be filtered out as long as the template doesn't return anything (which is the case when you wrap the template inside an if condition independently of what variables you define outside the if condition)

**Review by VictorSanh — APPROVED** [review]
lgtm, thank you @josrozen !
please update the spreadsheet, merging now

**Pull request merged**